### PR TITLE
[no ticket][risk=no] e2e minor improvement

### PR DESCRIPTION
**Pull request by aweng98**
Minor e2e test code refactor to reduce flakiness.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: e2e/app/component/workspace-card.ts [hunk 1]
```diff
@@ -69,14 +68,6 @@ export default class WorkspaceCard extends CardBase {
     return allCards;
   }
 
-  static async findAnyCard(page: Page): Promise<WorkspaceCard> {
```
Comment (aweng98) [inline e2e/app/component/workspace-card.ts:72]: Unused.

File: e2e/app/component/workspace-card.ts [hunk 2]
```diff
@@ -190,14 +173,7 @@ export default class WorkspaceCard extends CardBase {
   private getWorkspaceNameXpath(workspaceName: string): string {
     return (
       WorkspaceCardSelector.cardRootXpath +
-      `//*[${WorkspaceCardSelector.cardNameXpath} and normalize-space(text())="${workspaceName}"]`
-    );
-  }
-
-  private workspaceNameLinkSelector(workspaceName: string): string {
```
Comment (aweng98) [inline e2e/app/component/workspace-card.ts:197]: Unused.

File: e2e/app/page/cohort-actions-page.ts [hunk 3]
```diff
@@ -19,12 +19,13 @@ export default class CohortActionsPage extends AuthenticatedPage {
   }
 
   async isLoaded(): Promise<boolean> {
-    await Promise.all([waitForDocumentTitle(this.page, PageTitle), waitWhileLoading(this.page)]);
+    await waitForDocumentTitle(this.page, PageTitle);
```
Comment (aweng98) [inline e2e/app/page/cohort-actions-page.ts:22]:
```
await Promise.all([waitForDocumentTitle(this.page, PageTitle), waitWhileLoading(this.page)]);
```
is a bad code. `waitWhileLoading()` should run after `waitForDocumentTitle()`.

File: e2e/app/page/notebook-page.ts [hunk 4]
```diff
@@ -51,24 +52,34 @@ export enum KernelStatus {
...
+      (css, name) => {
+        const element = document.querySelector(css);
+        return element && element.textContent === name;
+      },
+      { timeout },
+      CssSelector.notebookName,
+      this.pageTitleRegex
+    );
+    await this.findRunButton(timeout);
+    await this.waitForKernelIdle(timeout);
```
Comment (aweng98) [inline e2e/app/page/notebook-page.ts:73]: Remove try/catch for `findRunButton` failure. Add notebook name check.

File: e2e/app/page/admin-workspace-page.ts [hunk 5]
```diff
@@ -27,7 +27,7 @@ export default class WorkspaceAdminPage extends AuthenticatedPage {
   }
 
   async isLoaded(): Promise<boolean> {
-    await Promise.all([waitForDocumentTitle(this.page, PageTitle)]);
```
Comment (aweng98) [inline e2e/app/page/admin-workspace-page.ts:30]: Don't need to use `Promise.all` since there is only one func call.

**Comment by aweng98** [comment]
CircleCI failed tests are not caused by this PR changes.

**Review by dolbeew — APPROVED** [review]

**Comment by aweng98** [comment]
Thank you.

**Pull request merged**